### PR TITLE
feature/LS-101 remerge changes

### PR DESCRIPTION
**Pull request by milomia**
no gdpr data

### CODE REVIEW

File: app/views/legal_services/journey/supplier_results.html.erb [hunk 1]
```diff
@@ -18,72 +9,72 @@
...
+        <%= govuk_form_group_with_optional_error(@journey, :lot1_region) do %>
+
+          <div class="ccs-search-results">
+            <ul class="ccs-results-list govuk-!-margin-bottom-7">
+
+              <% firms.each do |firm| %>
+                <% lot_no_slash = firm.gsub('/', '_') %>
+                <% lot_no_space = lot_no_slash.gsub(' ', '_') %>
+                <% detail =  LegalServices::Details.for_sme(firm) %>
+                <% sme_flag = detail.first.Is_an_SME %>
```
Comment (kennyevil) [inline app/views/legal_services/journey/supplier_results.html.erb]: sme should probably be a boolean attribute on the supplier model

File: app/models/legal_services/details.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    def self.all_firms
+      all.map(&:firm)
+    end
+
+    def self.all_regions
+      all.map(&:region)
+    end
+  end
+
+  Details.load_csv('legal_service/supplierdetails.csv')
```
Comment (kennyevil) [inline app/models/legal_services/details.rb]: What are you doing here? Suppliers are meant to be persisted. We can use static records for lots, services and regions but not suppliers. Also why are you using the word firm? The term we have used all over the place is supplier.

Comment (milomia) [reply app/models/legal_services/details.rb]: firm is the heading provided in the headings of the data - was not aware of supply teachers of management consulting naming convention as have not really looked at these.

File: app/views/legal_services/journey/regional_legal_service.html.erb [hunk 3]
```diff
@@ -54,20 +54,21 @@
 
               <div class="govuk-checkboxes ccs-govuk-checkboxes--thin">
 
-                <% lot1 = ['North East', 'North West','Yorkshire and the Humber','East Midlands', 'West Midlands','East of England', 'Greater London', 'South East','South West','Wales','Scotland','Northern Ireland'] %>
-
-                <% lot1.each do |region| %>
+                <% region1 = [['North East','NE'], ['North West','NW'],['Yorkshire and the Humber','YH'],['East Midlands','EM'], ['West Midlands','WM'],['East of England','EE'], ['Greater London','GL'], ['South East','SE'],['South West','SW'],['Wales','WA'],['Scotland','SC'],['Northern Ireland','NI']] %>
```
[the remark this answers is not in the document]
Comment (kennyevil) [reply app/views/legal_services/journey/regional_legal_service.html.erb]: This hasn't been addressed

[the remark this answers is not in the document]
Comment (milomia) [reply app/views/legal_services/journey/regional_legal_service.html.erb]: applied changes - removed existing results and put in dummy results page

File: app/views/legal_services/journey/supplier_results.html.erb [hunk 4]
```diff
@@ -18,72 +9,72 @@
...
+      <%sqlHash = {} %>
+      <% params[:lot1].each do | arg | %>
+        <% arg_no_slash = arg.gsub('/', '_') %>
+        <% arg_no_space = arg_no_slash.gsub(' ', '_') %>
+        <% arg = arg_no_space.downcase %>
+        <% sqlStr = "{:#{arg}=>'x'}" %>
+        <% sqlHash1 = eval(sqlStr) %>
+        <% sqlHash = sqlHash.merge(sqlHash1) %>
+      <%end %>
+      <% sqlStr ="{:region=>#{params[:region1]}}" %>
```
[the remark this answers is not in the document]
Comment (milomia) [reply app/views/legal_services/journey/supplier_results.html.erb]: cheers - forgot

File: app/views/legal_services/journey/supplier_results.html.erb [hunk 5]
```diff
@@ -18,72 +9,72 @@
...
+      <% sqlHash1 = eval(sqlStr) %>
```
[the remark this answers is not in the document]
Comment (milomia) [reply app/views/legal_services/journey/supplier_results.html.erb]: removed and replaced with dummy results page

File: app/views/legal_services/journey/supplier_results.html.erb [hunk 6]
```diff
@@ -1,12 +1,3 @@
-<%= add_optional_error_prefix_to_page_title(@journey.errors) %>
-
-
-
-      <%= hidden_fields_for_previous_steps_and_responses(@journey) %>
-
-
-
-
 <div class="govuk-grid-row">
```
[the remark this answers is not in the document]
Comment (milomia) [reply app/views/legal_services/journey/supplier_results.html.erb]: will do

**Pull request closed without merging**